internal/zinc-apiinfo/src/test/scala/sbt/internal/inc/ClassCanonicalNameSpec.scala

```diff
@@ -24,14 +24,15 @@ class ClassCanonicalNameSpec extends FlatSpec with Matchers {
   def checkRef(expected: Expected, x: AnyRef) = checkClass(expected, x.getClass, "ref")
   def checkClass(expected: Expected, c: Class[_], classSource: String) = {
     import expected._
-    it should f"for $nesting%-5s ($classSource) return $canonicalClassName" in assert(
-      getCustomCanonicalName(c) === canonicalClassName &&
-        getNativeCanonicalName(c) === nativeCanonicalClassName
-    )
+    it should f"for $nesting%-5s ($classSource) return $canonicalClassName" in {
+      assert(getCustomCanonicalName(c) === canonicalClassName)
+
+      // c.getCanonicalName is JDK implementation specific
+      // assert(getNativeCanonicalName(c) === nativeCanonicalClassName)
```

Comment:
We use mostly these names to be able to find the class file counterpart of a symbol. Could we add some tests that check the names of generated class files (by scalac) match what we derive from the analysis symbols? This should be JDK-independent.

Reply:
As far as I can tell ClassToAPI is invoked only by `AnalyzingJavaCompiler`, so the scalac encoding of class names I don't think matters much to the accuracy of ClassToAPI. In that sense the whole ClassCanonicalNameSpec is not very relevant imo since it's exercising `Class[_]` -> canonical name capability based on Scala-generated classes, and not Java. WDYT?

Reply:
I think you make a good point, it's not necessary to do that, so we can just merge this PR and rebase the other one based on it. 👍